cfme/services/catalogs/orchestration_template.py

```diff
@@ -1,166 +1,268 @@
-                'select_template':
-                [
-                    lambda ctx: accordion_tree('All Orchestration Templates',
-                        ctx['template_type'], ctx['template_name']),
-                    {'edit_template': lambda _: cfg_btn("Edit this Orchestration Template"),
-                     'create_service_dialog': lambda _: cfg_btn("Create Service Dialog from "
-                        "Orchestration Template")}
-                ],
-                'orch_template_type':
-                [
-                    lambda ctx: accordion_tree(
-                        'All Orchestration Templates', ctx['template_type']),
-                    {'create_new_template': lambda _: cfg_btn("Create new Orchestration Template")}
-                ]
-            }
-        ]
-    }
-)
-
-
-class OrchestrationTemplate(Updateable, Pretty):
-    pretty_attrs = ['template_type', 'template_name']
-
-    def __init__(self, template_type=None, template_name=None, description=None):
+from utils.appliance import Navigatable
+from utils.appliance.implementations.ui import navigator, CFMENavigateStep, navigate_to
+
+
+from . import ServicesCatalogView
+
+
+class OrchestrationTemplatesView(ServicesCatalogView):
+    title = Text("#explorer_title_text")
+
+    @property
+    def is_displayed(self):
+        return (
+            super(OrchestrationTemplatesView, self).is_displayed and
+            self.title.text == 'All Orchestration Templates' and
+            self.orchestration_templates.is_opened and
+            self.orchestration_templates.tree.currently_selected == ["All Orchestration Templates"])
+
+
+class CopyTemplateForm(ServicesCatalogView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    description = Input(name="description")
+    draft = Checkbox(name='draft')
+    content = ScriptBox(locator="//pre[@class=' CodeMirror-line ']/span")
+
+    cancel_button = Button('Cancel')
+
+
+class TemplateForm(CopyTemplateForm):
+
+    template_type = BootstrapSelect("type")
+
+
+class AddTemplateView(TemplateForm):
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Adding a new Orchestration Template" and
+            self.orchestration_templates.is_opened
+        )
+
+
+class EditTemplateView(TemplateForm):
+
+    save_button = Button('Save')
+    reset_button = Button('Reset')
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == "Editing {}".format(self.obj.name) and
+            self.orchestration_templates.is_opened
+        )
+
+
+class CopyTemplateView(CopyTemplateForm):
+
+    add_button = Button("Add")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.is_displayed and
+            self.title.text == "Copying {}".format(self.obj.name) and
+            self.orchestration_templates.is_opened
+        )
+
+
+class DetailsTemplateView(TemplateForm):
```

Comment:
It doesn't seem quite right to inherit TemplateForm here, since the details page in the UI isn't form elements. Should this inherit ServicesCatalogView with locators for the non-form fields?